cristin-project/src/main/java/no/unit/nva/cristin/projects/common/CristinQueryBuilder.java

```diff
@@ -0,0 +1,613 @@
+        cristinQuery.setValue(PAGE_CURRENT, page);
+        return this;
+    }
+
+    /**
+     * Items per page.
+     */
+    public CristinQueryBuilder withItemsPerPage(String itemsPerPage) {
+        cristinQuery.setValue(PAGE_ITEMS_PER_PAGE, itemsPerPage);
+        return this;
+    }
+
+    /**
+     * Setter sorting on 'start_date' and/or 'end_date'.
+     */
+    public CristinQueryBuilder withItemSort(String sort) {
+        cristinQuery.setValue(PAGE_SORT, sort);
+        return this;
+    }
+
+    /**
+     * Setter search by keyword.
+     */
+    public CristinQueryBuilder withKeyword(String keyword) {
+        cristinQuery.setValue(PROJECT_KEYWORD, keyword);
+        return this;
+    }
+
+    /**
+     * Preferred language.
+     */
+    public CristinQueryBuilder withLanguage(String language) {
+        var langOrDefault = nonNull(language)
+            ? language
+            : DEFAULT_LANGUAGE_CODE;
+        cristinQuery.setValue(LANGUAGE, langOrDefault);
+        return this;
+    }
+
+    /**
+     * Setter how many levels down from 'parent_unit_id' will be included in the search.
+     */
+    public CristinQueryBuilder withLevels(String levels) {
+        cristinQuery.setValue(LEVELS, levels);
+        return this;
+    }
+
+    /**
+     * Setter only those projects that have been modified since this date will be returned.
+     */
+    public CristinQueryBuilder withModifiedSince(String modifiedSince) {
+        cristinQuery.setValue(PROJECT_MODIFIED_SINCE, modifiedSince);
+        return this;
+    }
+
+    /**
+     * Setter Name.
+     */
+    public CristinQueryBuilder withName(String name) {
+        cristinQuery.setValue(NAME, name);
+        return this;
+    }
+
+    /**
+     * Setter Organization.
+     */
+    public CristinQueryBuilder withOrganization(String organization) {
+        if (nonNull(organization)) {
+            cristinQuery.setValue(PROJECT_ORGANIZATION, organization);
+        } else {
+            cristinQuery.setValue(QUERY, EMPTY_STRING);
+        }
+        return this;
+    }
+
+    /**
+     * Setter a participant of the project by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withParentUnitId(String parentUnitId) {
+        if (nonNull(parentUnitId)) {
+            var unitId = parentUnitId;
+            if (parentUnitId.matches(PATTERN_IS_URL)) {
+                unitId = getUnitIdFromOrganization(parentUnitId);
+            }
+            cristinQuery.setPath(PATH_ORGANISATION, unitId);
+        }
+        return this;
+    }
+
+    /**
+     * Setter a participant of the project by Cristin id, name or part of the name.
+     */
+    public CristinQueryBuilder withParticipant(String participant) {
+        cristinQuery.setValue(PROJECT_PARTICIPANT, participant);
+        return this;
+    }
+
+    /**
+     * Setter Identity.
+     */
+    public CristinQueryBuilder withPathIdentity(String identity) {
```

Comment:
What does this do?

Reply:
In RequestInfo, project Id or Organization Id is always refereed to as identity, 
I have prefixed path variables with path, in order to hint where it originate from, it was mostly for my own sanity